File: backend/src/modules/subscriptions/dto/cancel-subscription.dto.ts
import { IsIn, IsNotEmpty } from 'class-validator';
import { ApiProperty } from '@nestjs/swagger';

export class CancelSubscriptionDto {
  @ApiProperty({
    enum: ['now', 'period_end'],
    description: 'When to cancel: immediately or at end of current period',
  })
  @IsNotEmpty()
  @IsIn(['now', 'period_end'])
  cancelAt: 'now' | 'period_end';
}

File: sdks/rust/src/api/types/cancel_subscription_dto.rs
pub use crate::prelude::*;

/// Request type for API operation
#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
pub struct CancelSubscriptionDto {
    /// When to cancel: immediately or at end of current period
    #[serde(rename = "cancelAt")]
    pub cancel_at: CancelSubscriptionDtoCancelAt,
}
